chia/util/struct_stream.py

```diff
@@ -13,48 +11,88 @@ def __trunc__(self) -> int:
         ...
 
 
+packing_strings: Dict[Tuple[int, bool], Optional[str]] = {
+    (1, False): "!B",
+    (1, True): "!b",
+    (2, False): "!H",
+    (2, True): "!h",
+    (4, False): "!L",
+    (4, True): "!l",
+    (8, False): "!Q",
+    (8, True): "!q",
+    (16, False): None,
+}
+
+
+def parse_metadata_from_name(cls: Type[_T_StructStream]) -> Type[_T_StructStream]:
```

Comment:
what do you think about adding a test for this function?

Reply:
If nothing else, the error cases should be exercised to make sure they work.  At which point I'll probably add some other tests too.